src/adapters/http/server.ts

```diff
@@ -0,0 +1,128 @@
+import Adapter from "../../lib/adapter.js";
+import GleeMessage from "../../lib/message.js";
+import http from "http";
+import { validateData } from "../../lib/util.js";
+import GleeError from "../../errors/glee-error.js";
+import * as url from "url";
+
+class HttpAdapter extends Adapter {
+  private res = new Map();
+
+  name(): string {
+    return "HTTP server";
+  }
+
+  async connect(): Promise<this> {
+    return this._connect();
+  }
+
+  async send(message: GleeMessage): Promise<void> {
+    return this._send(message);
+  }
+
+  _connect(): Promise<this> {
+    return new Promise(async (resolve, reject) => {
+      const config = await this.resolveProtocolConfig("http");
+      const httpOptions = config?.server;
+      const serverUrl = new URL(this.serverUrlExpanded);
+      const httpServer = httpOptions?.httpServer || http.createServer();
+      const asyncapiServerPort = serverUrl.port || 80;
+      const optionsPort = httpOptions?.port;
+      const port = optionsPort || asyncapiServerPort;
+
+      httpServer.on("request", (req, res) => {
+        res.setHeader("Content-Type", "application/json");
+        const bodyBuffer = [];
+        let body: object;
+        req.on("data", (chunk) => {
+          bodyBuffer.push(chunk);
+        });
+        req.on("end", () => {
+          body = JSON.parse(Buffer.concat(bodyBuffer).toString());
+
+          this.res.set(this.serverName, res);
+          let { pathname } = new URL(req.url, serverUrl);
+          pathname = pathname.startsWith("/")
+            ? pathname.substring(1)
+            : pathname;
+          if (!this.parsedAsyncAPI.channel(pathname)) {
+            res.end("HTTP/1.1 404 Not Found1\r\n\r\n");
+            const err = new Error(
+              `A client attempted to connect to channel ${pathname} but this channel is not defined in your AsyncAPI file. here`
+            );
+            this.emit("error", err);
+            return reject(err);
+          }
+          const { query: query } = url.parse(req.url, true);
+          const searchParams = {
+            query,
+            body,
+          };
+          const httpChannelBinding = this.parsedAsyncAPI
+            .channel(pathname)
+            .binding("http");
+          if (httpChannelBinding) {
+            const { query, body ,method } = httpChannelBinding;
+            if (method && req.method !== method) {
+              this.emit("error", new Error(`Cannot ${req.method} ${pathname}`));
+              res.end("HTTP/1.1 400 Bad Request\r\n\r\n");
+              return;
+            }
+            if (query) {
+              const { isValid, humanReadableError, errors } = validateData(
+                searchParams.query,
+                query
+              );
+              if (!isValid) {
+                const err = new GleeError({ humanReadableError, errors });
+                this.emit("error", err);
+                res.end("HTTP/1.1 400 Bad Request\r\n\r\n");
```

Comment:
I think instead of sending only `Bad Request` we should also send a formated version of the `humanReadableError` object as it will help the user fix the issue faster.